fix(build): conditional exports so CJS TypeScript consumers can resolve subpath types

Surfaced while empirically validating INTEGRATION.md framework snippets (see PR #112). Runtime is fine; TypeScript is the problem.

## Symptom

A CJS-mode TypeScript project (`package.json` without `"type": "module"`, `tsconfig.json` with `"module": "Node16"` / `"NodeNext"` + matching `moduleResolution`) importing any oav subpath hits TS1479:

```
server.ts: error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("@aahoughton/oav")' call instead.
```

Reproduced at `/tmp/oav-integration-validation/cjs/` against both `@aahoughton/oav` and `@aahoughton/oav/spec`.

**Runtime works fine.** A pure-CJS Node.js script (no TypeScript) can `require('@aahoughton/oav')` and its subpaths — Node's resolution uses the `require` branch of the exports map correctly. The issue is TypeScript-only.

## Root cause

The `exports` maps in `packages/oav/package.json` and the root `package.json` use a flat `"types"` entry:

```json
".": {
  "types": "./dist/index.d.ts",
  "import": "./dist/index.js",
  "require": "./dist/index.cjs"
}
```

TypeScript infers module kind from type-file extension: `.d.ts` → ESM, `.d.cts` → CJS. With a single flat `"types": "./dist/index.d.ts"`, TypeScript treats every import of the package as ESM, regardless of which branch Node runtime actually resolves. A CJS consumer then gets TS1479.

tsup already produces the `.d.cts` sibling files on every entry (confirmed in `packages/oav/dist/`) — they're just not wired into the exports map.

## Fix

Conditional exports per the Node16 pattern, for every entry in both package.jsons. For each entry `X`:

```json
"./X": {
  "import": { "types": "./dist/X.d.ts",  "default": "./dist/X.js"  },
  "require": { "types": "./dist/X.d.cts", "default": "./dist/X.cjs" }
}
```

Entries to update:
- `packages/oav/package.json`: `.`, `./schema`, `./schema/internals`, `./spec`, `./formats`, `./core`, `./validator/internals`
- Root `package.json` (`@aahoughton/oav-core`): same set

The top-level `"types"` field can stay (legacy TS compatibility).

## Adjacent note

TypeScript projects on the older `moduleResolution: Node10` can't resolve subpaths like `@aahoughton/oav/spec` at all — Node10 predates the `exports` map. The workaround is to upgrade to `Node16` / `NodeNext` / `Bundler`. Worth a line in the INTEGRATION.md adapter prerequisites once the fix lands.

## Test

`/tmp/oav-integration-validation/cjs/server.ts` reproduces the error; same file with the conditional exports fix in place should type-check cleanly.

--- a/package.json
+++ b/package.json
@@ -38,39 +38,74 @@
   "types": "./dist/index.d.ts",
   "exports": {
     ".": {
-      "types": "./dist/index.d.ts",
-      "import": "./dist/index.js",
-      "require": "./dist/index.cjs"
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      },
+      "require": {
+        "types": "./dist/index.d.cts",
+        "default": "./dist/index.cjs"
+      }
     },
     "./schema": {
-      "types": "./dist/schema.d.ts",
-      "import": "./dist/schema.js",
-      "require": "./dist/schema.cjs"
+      "import": {
+        "types": "./dist/schema.d.ts",
+        "default": "./dist/schema.js"
+      },
+      "require": {
+        "types": "./dist/schema.d.cts",
+        "default": "./dist/schema.cjs"
+      }
     },
     "./schema/internals": {
-      "types": "./dist/schema-internals.d.ts",
-      "import": "./dist/schema-internals.js",
-      "require": "./dist/schema-internals.cjs"
+      "import": {
+        "types": "./dist/schema-internals.d.ts",
+        "default": "./dist/schema-internals.js"
+      },
+      "require": {
+        "types": "./dist/schema-internals.d.cts",
+        "default": "./dist/schema-internals.cjs"
+      }
     },
     "./spec": {
-      "types": "./dist/spec.d.ts",
-      "import": "./dist/spec.js",
-      "require": "./dist/spec.cjs"
+      "import": {
+        "types": "./dist/spec.d.ts",
+        "default": "./dist/spec.js"
+      },
+      "require": {
+        "types": "./dist/spec.d.cts",
+        "default": "./dist/spec.cjs"
+      }
     },
     "./formats": {
-      "types": "./dist/formats.d.ts",
-      "import": "./dist/formats.js",
-      "require": "./dist/formats.cjs"
+      "import": {
+        "types": "./dist/formats.d.ts",
+        "default": "./dist/formats.js"
+      },
+      "require": {
+        "types": "./dist/formats.d.cts",
+        "default": "./dist/formats.cjs"
+      }
     },
     "./core": {
-      "types": "./dist/core.d.ts",
-      "import": "./dist/core.js",
-      "require": "./dist/core.cjs"
+      "import": {
+        "types": "./dist/core.d.ts",
+        "default": "./dist/core.js"
+      },
+      "require": {
+        "types": "./dist/core.d.cts",
+        "default": "./dist/core.cjs"
+      }
     },
     "./validator/internals": {
-      "types": "./dist/validator-internals.d.ts",
-      "import": "./dist/validator-internals.js",
-      "require": "./dist/validator-internals.cjs"
+      "import": {
+        "types": "./dist/validator-internals.d.ts",
+        "default": "./dist/validator-internals.js"
+      },
+      "require": {
+        "types": "./dist/validator-internals.d.cts",
+        "default": "./dist/validator-internals.cjs"
+      }
     },
     "./package.json": "./package.json"
   },

--- a/packages/oav/package.json
+++ b/packages/oav/package.json
@@ -42,39 +42,74 @@
   "types": "./dist/index.d.ts",
   "exports": {
     ".": {
-      "types": "./dist/index.d.ts",
-      "import": "./dist/index.js",
-      "require": "./dist/index.cjs"
+      "import": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      },
+      "require": {
+        "types": "./dist/index.d.cts",
+        "default": "./dist/index.cjs"
+      }
     },
     "./schema": {
-      "types": "./dist/schema.d.ts",
-      "import": "./dist/schema.js",
-      "require": "./dist/schema.cjs"
+      "import": {
+        "types": "./dist/schema.d.ts",
+        "default": "./dist/schema.js"
+      },
+      "require": {
+        "types": "./dist/schema.d.cts",
+        "default": "./dist/schema.cjs"
+      }
     },
     "./schema/internals": {
-      "types": "./dist/schema-internals.d.ts",
-      "import": "./dist/schema-internals.js",
-      "require": "./dist/schema-internals.cjs"
+      "import": {
+        "types": "./dist/schema-internals.d.ts",
+        "default": "./dist/schema-internals.js"
+      },
+      "require": {
+        "types": "./dist/schema-internals.d.cts",
+        "default": "./dist/schema-internals.cjs"
+      }
     },
     "./spec": {
-      "types": "./dist/spec.d.ts",
-      "import": "./dist/spec.js",
-      "require": "./dist/spec.cjs"
+      "import": {
+        "types": "./dist/spec.d.ts",
+        "default": "./dist/spec.js"
+      },
+      "require": {
+        "types": "./dist/spec.d.cts",
+        "default": "./dist/spec.cjs"
+      }
     },
     "./formats": {
-      "types": "./dist/formats.d.ts",
-      "import": "./dist/formats.js",
-      "require": "./dist/formats.cjs"
+      "import": {
+        "types": "./dist/formats.d.ts",
+        "default": "./dist/formats.js"
+      },
+      "require": {
+        "types": "./dist/formats.d.cts",
+        "default": "./dist/formats.cjs"
+      }
     },
     "./core": {
-      "types": "./dist/core.d.ts",
-      "import": "./dist/core.js",
-      "require": "./dist/core.cjs"
+      "import": {
+        "types": "./dist/core.d.ts",
+        "default": "./dist/core.js"
+      },
+      "require": {
+        "types": "./dist/core.d.cts",
+        "default": "./dist/core.cjs"
+      }
     },
     "./validator/internals": {
-      "types": "./dist/validator-internals.d.ts",
-      "import": "./dist/validator-internals.js",
-      "require": "./dist/validator-internals.cjs"
+      "import": {
+        "types": "./dist/validator-internals.d.ts",
+        "default": "./dist/validator-internals.js"
+      },
+      "require": {
+        "types": "./dist/validator-internals.d.cts",
+        "default": "./dist/validator-internals.cjs"
+      }
     },
     "./package.json": "./package.json"
   },

